README.md

```diff
@@ -1,2 +1,269 @@
 # pyvaultone
-Project to abstract vault one interaction.
+
+[![Build Status](https://travis-ci.com/NeowayLabs/pyvaultone.svg?branch=main)](https://travis-ci.com/NeowayLabs/pyvaultone)
+
+This project is a library to abstract relevant interactions to the VaultOne application API. Relevant interactions are defined as necessary interactions in order for a bot to retrieve credentials that will be used in the crawling process.
+
+You need a valid Vault One credential to use this library. The username and the password must be encoded in base64 format and must be set to the following environment variables: VAULT_ONE_API_USERNAME, VAULT_ONE_API_PASSWORD and VAULT_ONE_API_BASE_URL.
+
+These interactions are read-only, thus this library will not implement write operations using the VaultOne API. The necessary API endpoints are the following:
+  - Obtaining an access token to the API;
+  - Retrieving credentials based on a filter that identifies the credential;
+  - Acquiring a credential secret using a credential unique id provided by the VaultOne API.
+
+The API endpoints that are implemented in this first version are the following, and will be referred hereinafter by their corresponding names:
+  - POST /api/TokenAuth/Authenticate - Authenticate Request
+  - GET /api/services/app/Credentials/GetAllCredentials?Filter= - Get Credentials Request
+  - GET /api/services/app/Credentials/GetCredentialSecret?CredentialId= - Get Credential Secret Request
+
+## Authenticate Request
```

Comment:
why do we need so much information about vault one API? I think that is better to explain only how to use this lib/functions... and put one link to vault one API docs...  @vavieira10

Reply:
@LeoCBS I removed the information that I thought it was unecessary. But most of information are helpful for the eventual lib users